src/cargo/util/lints.rs

```diff
@@ -880,6 +882,111 @@ pub fn unused_dependencies(
     Ok(())
 }
 
+pub fn unexpected_target_cfgs(
+    ws: &Workspace<'_>,
+    pkg: &Package,
+    path: &Path,
+    error_count: &mut usize,
+    gctx: &GlobalContext,
+) -> CargoResult<()> {
+    fn warn_on_unexpected_cfgs(
+        gctx: &GlobalContext,
+        check_cfg: &CheckCfg,
+        cfg_expr: &CfgExpr,
+        lint_level: LintLevel,
+        error_count: &mut usize,
+        path: Option<&Path>,
+        suffix: &str,
+    ) -> CargoResult<()> {
+        let prefix = if let Some(path) = path {
+            format!("{path}: ", path = path.display())
+        } else {
+            "".to_string()
+        };
```

Comment:
Why does this deal with `Option<Path>` and why are we creating this before we even know if we have a lint?

Reply:
It was coming from a previous iteration. Fixed.